### DISCUSSION TITLE
Suggish Response to Remote Control Commands

### DISCUSSION THREAD

**Discussion by bill-lj** [GerryDazoo/Slinger#253] [Q&A] [answered]
Slinger is up and running.
I was hoping for some insight into speeding up remote control functions, specially FF/WW/Pause/Stop/Play.
At Gerry's suggestion, I've tried multiple players, e.g. SMPlayer, OBS, VLC without success.
Don't know if newest release addresses this, but any pointers/suggestions would be appreciated.

**Comment by AndyJStubbs** [comment]
If your slingbox is in the same location that you are streaming from the fastest way to speed up the remote commands is by using a cheap  IR repeater and a spare remote control. The latency from encoding and sending remote commands even using slings own software was a bit tedious at best unfortunately.

**Comment by GerryDazoo** [comment, marked as answer]
The issue isn't the slinger software. If you watch the source and not the stream you'll see the IR commands are quite responsive. The delay from the time you press the button to the time the slingbox gets the message is worst case 500 milliseconds.The issue is how much buffering your media client is doing. Out of the box VLC adds at least 4 seconds to the latency. So you need to work on your media players' configuration. I've been using ffPlayer with minimal buffering and it's usable.
Andy: Buying a cheap IR repeater will not solve the latency introduced by the buffering in the MediaPlayer.

**Reply by AndyJStubbs** [reply]
Good point - I hadn't factored in the latency added by the media player stream itself

**Reply by GerryDazoo** [reply]
Next release the server latency will be in the 10's of mSec.

**Reply by AndyJStubbs** [reply]
Nice!

**Comment by garabik** [comment]
> Slinger is up and running. I was hoping for some insight into speeding up remote control functions, specially FF/WW/Pause/Stop/Play. At Gerry's suggestion, I've tried multiple players, e.g. SMPlayer, OBS, VLC without success. Don't know if newest release addresses this, but any pointers/suggestions would be appreciated.

A bit late reply (I have just resurrected my SlingBox and set up slinger), but posting it anyway, in hope it might help if someone searches the answers: I am using this command (thanks to https://github.com/mpv-player/mpv/issues/4213) on Linux and the improvement is enormous, almost like an original v2 SlingPlayer with UDP (which I suspect flushes the buffer when you start using the virtual remote):

    ffplay http://127.0.0.1:8080/slingbox -nostats  -flags low_delay  -fflags nobuffer+fastseek+flush_packets -analyzeduration 0 -sync ext -af aresample=async=1:min_comp=0.1

(it still remains a PITA to use the remote extensively, e.g. to organize a huge channel list or something - but at least it is usable)
